app/models/user.rb

```diff
@@ -2,6 +2,14 @@ class User < ApplicationRecord
   has_many :user_videos, dependent: :destroy
   has_many :videos, through: :user_videos
 
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships
+  has_many :inverse_friendships,
```

Comment:
not sure what inverse relationships are, i'll have to look this up

Reply:
just checked out the docs on this one, this is pretty cool, nice work!

Reply:
supposedly its suppose to be like if I make a friendship with someone then they also make a friendship with me since both sides are a user

Reply:
right yeah because technically everybody could be a user and and everybody could be a friend for a different user

Reply:
that's a mind bender haha